nautobot/ipam/models.py

```diff
@@ -265,9 +265,15 @@ def _deconstruct_prefix(self, pre):
         if pre:
             if isinstance(pre, str):
                 pre = netaddr.IPNetwork(pre)
-            # if |address.prefixlen| is 32 (ip4) or 128 (ip6)
-            # then |address.broadcast| is None
-            broadcast = pre.broadcast if pre.broadcast else pre.network
+            # Note that our "broadcast" field is actually the last IP address in this prefix.
+            # This is different from the more accurate technical meaning of a network's broadcast address in 2 cases:
+            # 1. For a point-to-point prefix (IPv4 /31 or IPv6 /127), there are two addresses in the prefix,
+            #    and neither one is considered a broadcast address. We store the second address as our "broadcast".
+            # 2. For a host prefix (IPv6 /32 or IPv6 /128) there's only one address in the prefix.
+            #    We store this address as both the network and the "broadcast".
+            # This variance is intentional in both cases as we use the "broadcast" primarily for filtering and grouping
+            # of addresses and prefixes, not for packet forwarding. :-)
+            broadcast = pre.broadcast if pre.broadcast else pre[-1]
```

Comment:
It occurs to me that I might need to add a migration to fix the "broadcast" value for /31 and /127 entries in the database that were entered prior to this fix. Thoughts?

Reply:
I was going to complain about the indexing, but realized it would just require an extra step if we used `pre.last` since that returns an `int`, not an `IPNetwork`.

Reply:
Confirmed - will add